src/expression.c

```diff
@@ -6104,8 +6104,9 @@ Expression *TypeidExp::semantic(Scope *sc)
     {
         /* Get the dynamic type, which is .classinfo
          */
-        e = new DotIdExp(ea->loc, ea, Id::classinfo);
-        e = e->semantic(sc);
+        ea = ea->semantic(sc);
+        e = new TypeidExp(ea->loc, ea);
+        e->type = Type::typeinfoclass->type;
```

Comment:
`typeid(I)` and `typeid(J)` return the pointer to Type::vtinfo variable, but `typeid(i)` and `i.classinfo` return the pointer that directly points `TypeInfo_Class` instance. If you change the `getTypeInfo` function in `e2ir.c` as follows, the asserts result will be changed.

```cpp
elem *getTypeInfo(Type *t, IRState *irs)
{
    assert(t->ty != Terror);
    genTypeInfo(t, NULL);
    if (t->ty == Tclass)    // test
    {
        //printf("getTypeInfo class = %s\n", t->toChars());
        elem *e = el_ptr(toSymbol(((TypeClass *)t)->sym));
        return e;
    }
    elem *e = el_ptr(toSymbol(t->vtinfo));
    return e;
}

```